File: common/src/main/java/org/sandboxpowered/silica/network/clientbound/Disconnect.java
package org.sandboxpowered.silica.network.clientbound;

import org.sandboxpowered.silica.network.Packet;
import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.server.SilicaServer;

public class Disconnect implements Packet {
    private String reason;

    public Disconnect(String reason) {
        this.reason = reason;
    }

    public Disconnect() {
    }

    @Override
    public void read(PacketByteBuf buf) {
        reason = buf.readString();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeString(reason);
    }

    @Override
    public void handle(PacketHandler packetHandler, SilicaServer server) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/vanilla/network/login/clientbound/Disconnect.kt
package org.sandboxpowered.silica.vanilla.network.login.clientbound

import org.sandboxpowered.silica.vanilla.network.Connection
import org.sandboxpowered.silica.vanilla.network.Packet
import org.sandboxpowered.silica.vanilla.network.PacketByteBuf
import org.sandboxpowered.silica.vanilla.network.PacketHandler

class Disconnect(private var reason: String?) : Packet {

    constructor() : this(null)

    override fun read(buf: PacketByteBuf) {
        reason = buf.readString()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeString(reason!!)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {}
}